server/src/main/java/org/opensearch/search/profile/query/QueryProfiler.java

```diff
@@ -89,10 +100,55 @@ public long stopAndAddRewriteTime() {
     }
 
     /**
+     * The rewriting process is complex and hard to display because queries can undergo significant changes.
+     * Instead of showing intermediate results, we display the cumulative time for the non-concurrent search case.
+     * In the concurrent search case, we compute the sum of the non-overlapping time across all slices and add
+     * the rewrite time from the non-concurrent path.
      * @return total time taken to rewrite all queries in this profile
      */
     public long getRewriteTime() {
-        return ((AbstractQueryProfileTree) profileTree).getRewriteTime();
+        long rewriteTime = ((AbstractQueryProfileTree) profileTree).getRewriteTime();
+        if (profileTree instanceof ConcurrentQueryProfileTree) {
+            List<Timer> timers = getConcurrentPathRewriteTimers();
+            LinkedList<long[]> mergedIntervals = mergeRewriteTimeIntervals(timers);
+            for (long[] interval : mergedIntervals) {
+                rewriteTime += interval[1] - interval[0];
+            }
+        }
+        return rewriteTime;
+    }
+
+    // package private for unit testing
+    LinkedList<long[]> mergeRewriteTimeIntervals(List<Timer> timers) {
```

Comment:
With separate `ConcurrentQueryProfiler` mechanism this custom logic can also be moved there